app/assets/javascripts/discourse/app/lib/screen-track.js

```diff
@@ -85,6 +89,99 @@ export default class {
     this._anonCallback = cb;
   }
 
+  consolidateTimings(timings, topicTime, topicId) {
+    this._consolidatedTimings = this._consolidatedTimings || [];
+
+    let found = this._consolidatedTimings.find((elem) => elem[2] === topicId);
+
+    if (found) {
+      this._consolidatedTimings.removeObject(found);
+
+      const oldTimings = found[0];
+      Object.keys(oldTimings).forEach((id) => {
+        if (timings[id]) {
+          oldTimings[id] += timings[id];
+        }
+      });
+      found[1] += topicTime;
+      found[0] = Object.assign({}, timings, found[0]);
+    }
+
+    found = found || [timings, topicTime, topicId];
+    this._consolidatedTimings.unshift(found);
+
+    return this._consolidatedTimings;
+  }
+
+  sendNextConsolidatedTiming() {
+    if (!this._consolidatedTimings || this._consolidatedTimings.length === 0) {
+      return Promise.resolve();
+    }
+
+    if (this._inProgress) {
+      return Promise.resolve();
+    }
+
+    if (
+      this._blockSendingToServerTill &&
+      this._blockSendingToServerTill > Date.now()
+    ) {
+      return Promise.resolve();
+    }
+
+    this._ajaxFailures = this._ajaxFailures || 0;
+
+    const item = this._consolidatedTimings.shift();
+    const timings = item[0];
+    const topicTime = item[1];
+    const topicId = item[2];
+
+    this._inProgress = true;
+
+    return ajax("/topics/timings", {
+      data: {
+        timings: timings,
+        topic_time: topicTime,
+        topic_id: topicId,
+      },
+      cache: false,
+      type: "POST",
+      headers: {
+        "X-SILENCE-LOGGER": "true",
+        "Discourse-Background": "true",
+      },
+    })
+      .then(() => {
+        this._ajaxFailures = 0;
+        const topicController = this._topicController;
+        if (topicController) {
+          const postNumbers = Object.keys(timings).map((v) => parseInt(v, 10));
+          topicController.readPosts(topicId, postNumbers);
+        }
+      })
+      .catch((e) => {
+        if (ALLOWED_AJAX_FAILURES.indexOf(e.jqXHR.status) >= -1) {
+          const delay = AJAX_FAILURE_DELAYS[this._ajaxFailures];
+          this._ajaxFailures += 1;
+
+          if (delay) {
+            this._blockSendingToServerTill = Date.now() + delay;
+            // we did not send to the server, got to re-queue it
+            this.consolidateTimings(timings, topicTime, topicId);
+          }
+        }
+        // TODO decide if we want to log this anywhere, at the moment we simply eat the
+        // errors and do not report anywhere (it may still show up on server logs though)
+        //
```

Comment:
We could console.warn/log here. It might be useful if we are trying to help someone debug something and they can open it up and see the errors.

Reply:
Sure, added. We have been eating these errors now for a couple of years but this code is more complicated so more debugging can help